mcstatus/protocol/connection.py

```diff
@@ -20,199 +22,501 @@
+        remaining = unsigned_int32(value).value
+        for _ in range(5):
+            if not remaining & -0x80:  # remaining & ~0x7F == 0:
+                self.write(struct.pack("!B", remaining))
+                if value > 2**31 - 1 or value < -(2**31):
+                    break
+                return
+            self.write(struct.pack("!B", remaining & 0x7F | 0x80))
+            remaining >>= 7
+        raise ValueError(f'The value "{value}" is too big to send in a varint')
+
+    def write_varlong(self, value: int) -> None:
+        """
+        Write varlong with value value to self.
+
+        :param value: Max 2 ** 63 - 1, min -(2 ** 63).
+        :raises ValueError: If value is out of range.
+        """
+        remaining = unsigned_int64(value).value
+        for _ in range(10):
+            if not remaining & -0x80:  # remaining & ~0x7F == 0:
+                self.write(struct.pack("!B", remaining))
+                if value > 2**63 - 1 or value < -(2**31):
+                    break
+                return
+            self.write(struct.pack("!B", remaining & 0x7F | 0x80))
+            remaining >>= 7
+        raise ValueError(f'The value "{value}" is too big to send in a varlong')
+
+    def write_utf(self, value: str) -> None:
+        """Write varint of length of value up to 32767 bytes, then write value encoded with utf8."""
+        self.write_varint(len(value))
+        self.write(bytearray(value, "utf8"))
+
+    def write_ascii(self, value: str) -> None:
+        """Write value encoded with ISO-8859-1, then write an additional 0x00 at the end."""
+        self.write(bytearray(value, "ISO-8859-1"))
+        self.write(bytearray.fromhex("00"))
+
+    def write_short(self, value: int) -> None:
+        """Write 2 bytes for value -32768 - 32767."""
+        self.write(self._pack("h", value))
+
+    def write_ushort(self, value: int) -> None:
+        """Write 2 bytes for value 0 - 65535 (2 ** 16 - 1)."""
+        self.write(self._pack("H", value))
+
+    def write_int(self, value: int) -> None:
+        """Write 4 bytes for value -2147483648 - 2147483647."""
+        self.write(self._pack("i", value))
+
+    def write_uint(self, value: int) -> None:
+        """Write 4 bytes for value 0 - 4294967295 (2 ** 32 - 1)."""
+        self.write(self._pack("I", value))
+
+    def write_long(self, value: int) -> None:
+        """Write 8 bytes for value -9223372036854775808 - 9223372036854775807."""
+        self.write(self._pack("q", value))
+
+    def write_ulong(self, value: int) -> None:
+        """Write 8 bytes for value 0 - 18446744073709551613 (2 ** 64 - 1)."""
+        self.write(self._pack("Q", value))
 
-    def _unpack(self, format: str, data: Union[BytesConvertable, SupportsBytes]) -> int:
-        return struct.unpack(">" + format, bytes(data))[0]
+    def write_buffer(self, buffer: "Connection") -> None:
+        """Flush buffer, then write a varint of the length of the buffer's
+        data, then write buffer data."""
```

Comment:
Our maximum line length is 127, so this can easily fit into a single docstring line.

```suggestion
        """Flush buffer, then write a varint of the length of the buffer's data, then write buffer data."""
```